core/src/replay_stage.rs

```diff
@@ -196,11 +201,13 @@ impl ReplayStage {
             snapshot_package_sender,
             block_commitment_cache,
             transaction_status_sender,
+            tower_snapshot_path,
         } = config;
 
         let (root_bank_sender, root_bank_receiver) = channel();
         trace!("replay stage");
-        let mut tower = Tower::new(&my_pubkey, &vote_account, &bank_forks.read().unwrap());
+        let mut tower =
+            Self::reload_tower(&my_pubkey, &tower_snapshot_path, &vote_account, &bank_forks);
```

Comment:
@sagar-solana, @TristanDebrunner 

Because replay always sets root in tower, then sets root in blocktree, I think we should check here that the root in this tower exists in blocktree's root column family. If it doesn't then insert the root into blocktree.

Reply:
Do we need to worry about having a bad saved tower state in some way? With that change, seems like it could inject a bad root into the blocktree column. But I guess that's maybe negligible compared to other havoc it could cause.

Reply:
Now with signing the saved state this shouldn't be a problem. Will add